experiment/manual-trigger/manual-trigger.go

```diff
@@ -0,0 +1,178 @@
+	flag.StringVar(&o.jenkinsBearerToken, "jbtoken", "", "Path to file containing Jenkins Bearer token.")
+
+	flag.StringVar(&o.jobName, "jobname", "", "Name of Jenkins job")
+
+	flag.IntVar(&o.num, "num", 0, "GitHub issue number")
+	flag.StringVar(&o.org, "org", "", "GitHub organization")
+	flag.StringVar(&o.repo, "repo", "", "GitHub repository")
+	flag.Parse()
+	return o
+}
+
+func sanityCheckFlags(o options) error {
+	if o.num <= 0 {
+		return fmt.Errorf("empty or invalid --num")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("empty --org")
+	}
+	if o.repo == "" {
+		return fmt.Errorf("empty --repo")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("empty --ghtoken")
+	}
+	if o.jenkinsBearerToken == "" {
+		return fmt.Errorf("empty --jbtoken")
+	}
+	if o.jenkinsBearerToken == "" {
+		return fmt.Errorf("empty --jobname")
+	}
+
+	_, err := url.Parse(o.githubEndpoint)
+	if o.githubEndpoint == "" || err != nil {
+		return fmt.Errorf("must specify a valid URL to --githubApiUrl")
+	}
+
+	_, err = url.Parse(o.jenkinsEndpoint)
+	if o.jenkinsEndpoint == "" || err != nil {
+		return fmt.Errorf("must specify a valid URL to --jenkinsurl")
+	}
+
+	return nil
+}
+
+func readToken(path string) (string, error) {
+	contents, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(contents)), nil
+}
+
+func main() {
+	o := flagOptions()
+	err := sanityCheckFlags(o)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	jbtoken, err := readToken(o.jenkinsBearerToken)
+	if err != nil {
+		log.Fatalf("cannot read file specified by --jbtoken %v", err)
+	}
+
+	ghtoken, err := readToken(o.githubToken)
+	if err != nil {
+		log.Fatalf("cannot read file specified by --ghtoken: %v", err)
+	}
+
+	auth := jenkins.BearerTokenAuthConfig{
+		Token: jbtoken,
+	}
+
+	authconfig := jenkins.AuthConfig{
+		Basic:       nil,
+		BearerToken: &auth,
+	}
+
+	jc := jenkins.NewClient(o.jenkinsEndpoint, &authconfig)
+
+	gc := github.NewClient(ghtoken, o.githubEndpoint)
+
+	fmt.Println("Querying GitHub for details...")
+	pr, err := gc.GetPullRequest(o.org, o.repo, o.num)
+	if err != nil {
+		log.Fatalf("Unable to get specified pull request %v", err)
+	}
+
+	ref, err := gc.GetRef(o.org, o.repo, "heads/"+pr.Base.Ref)
+	if err != nil {
+		log.Fatalf("Unable to get reference to pull request %v", err)
+	}
+	fmt.Println("done")
+
+	refs := kube.Refs{
+		Org:     o.org,
+		Repo:    o.repo,
+		BaseRef: pr.Base.Ref,
+		BaseSHA: ref,
+		Pulls: []kube.Pull{
+			{
+				Number: pr.Number,
+				Author: pr.User.Login,
+				SHA:    pr.Head.SHA,
+			},
+		},
+	}
+
+	spec := kube.ProwJobSpec{
+		Type: kube.PresubmitJob,
+		Job:  o.jobName,
+		Refs: refs,
+
+		Report:         false,
+		Context:        "",
+		RerunCommand:   "",
+		MaxConcurrency: 1,
+	}
+
+	prowjob := pjutil.NewProwJob(spec)
+
+	fmt.Println("Submitting to Jenkins...")
+	err = jc.Build(&prowjob)
+	if err != nil {
+		log.Fatalf("Error submitting to Jenkins %v", err)
+	}
+	fmt.Println("done")
```

Comment:
For instance, you may want to print out the job name/number and the parameters that were passed here.